README.md

```diff
@@ -90,6 +86,41 @@ try {
 }
 ```
 
+
+### Using 2-legged OAuth2 using Client Credentials
+
+The Client Credentials grant is used when applications request an access token to access their own resources, not on behalf of a user.
+
+```php
+<?php
+
+// Bootup the Composer autoloader
+include __DIR__ . '/vendor/autoload.php';  
+
+use Mautic\Auth\ApiAuth;
+
+$settings = [
+    'AuthMethod'   => 'TwoLeggedOAuth2',
+    'clientKey'    => '',
+    'clientSecret' => '',
+    'baseUrl'      => '',
+];
+
+// $settings['accessToken'] = 'your stored access token';
+
+
+$initAuth = new ApiAuth();
+$auth     = $initAuth->newAuth($settings, $settings['AuthMethod']);
+
+if (!isset($settings['accessToken'])) {
+    // store it for one hour and use it in $settings above
+    $accessToken = $auth->getAccessToken();
```

Comment:
In the regular `OAuth` provider, there's a function called `getAccessTokenData()` which returns an array with `access_token, expires, token_type, refresh_token`. 

Could we have the same method for `TwoLeggedOAuth2` please, to keep things consistent? It can return `access_token, expires, token_type`. Especially `expires` is interesting here, because folks can use it to store when the token expires and they need to renew it 😊 

You could store the expiration time just like it was done in `lib/Auth/OAuth.php` 😊 

https://github.com/mautic/api-library/blob/9087ddec6400eb0c8aa43859eb7710cc9948c1ce/lib/Auth/OAuth.php#L679